components/__tests__/SingleSelectWithFilter-test.js

```diff
@@ -0,0 +1,63 @@
+/* @jest-environment jsdom */
+
+import React from 'react';
+import { shallow, mount } from 'enzyme';
+import { Text } from 'react-native';
+import SingleSelectWithFilter from '../SingleSelectWithFilter';
+import ModalFilterPicker from 'react-native-modal-filter-picker';
+
+describe('single select with filter', () => {
+
+  it('sets items', () => {
+    const wrapper = shallow(<SingleSelectWithFilter
+      items= {[
+        { key: 4, label: 'First test' }, 
+        { key: 3, label: 'Second test' }, 
+        { key: 2, label: 'Third text' }
+      ]}
+      selectedItem= {`item-4`}
+      />);
+
+  expect(wrapper
+    .find(ModalFilterPicker).first()
+    .prop('options')).toEqual([
+      { key: 4, label: 'First test' }, 
+      { key: 3, label: 'Second test' }, 
+      { key: 2, label: 'Third text' }]);
+  });
+  
+  it('sets selected item', () => {
+    const wrapper = shallow(<SingleSelectWithFilter
+      items= {[
+        { key: 4, label: 'First test' }, 
+        { key: 3, label: 'Second test' }, 
+        { key: 2, label: 'Third text' }
+      ]}
+      selectedItem= {`item-4`}
+      />);
+     
+  expect(wrapper
+    .find(Text).first()
+    .prop('children')).toEqual('First test');
+  });
+
+  it('Unselect', () => {
```

Comment:
This test is not actually unselecting anything. It is opening the selection modal and then closing it. It should be renamed to match. Example: "opens and closes selection modal"